flink-kubernetes/src/test/java/org/apache/flink/kubernetes/kubeclient/TestingFlinkKubeClient.java

```diff
@@ -103,6 +121,52 @@ public KubernetesWatch watchPodsAndDoCallback(Map<String, String> labels, WatchC
 		return watchPodsAndDoCallbackFunction.apply(labels, podCallbackHandler);
 	}
 
+	@Override
+	public CompletableFuture<Void> createConfigMap(KubernetesConfigMap configMap) {
+		configMapStore.putIfAbsent(configMap.getName(), configMap);
+		return CompletableFuture.completedFuture(null);
+	}
+
+	@Override
+	public Optional<KubernetesConfigMap> getConfigMap(String name) {
+		final KubernetesConfigMap configMap = configMapStore.get(name);
+		if (configMap == null) {
+			return Optional.empty();
+		}
+		return Optional.of(new MockKubernetesConfigMap(configMap.getName(), new HashMap<>(configMap.getData())));
+	}
+
+	@Override
+	public CompletableFuture<Boolean> checkAndUpdateConfigMap(
+			String configMapName,
+			Predicate<KubernetesConfigMap> checker,
+			FunctionWithException<KubernetesConfigMap, KubernetesConfigMap, ?> function) {
+		return getConfigMap(configMapName).map(FunctionUtils.uncheckedFunction(
+			configMap -> {
+				final boolean shouldUpdate = checker.test(configMap);
+				if (shouldUpdate) {
+					configMapStore.put(configMap.getName(), function.apply(configMap));
+				}
+				return CompletableFuture.completedFuture(shouldUpdate);
+			}))
+			.orElseThrow(() -> new FlinkRuntimeException("ConfigMap " + configMapName + " not exists."));
+	}
```

Comment:
Since all the new introduced operations are ConfigMap related. So I add a new field `configMapStore` to manage the state. It is more natural to create/update/read/delete the ConfigMap state.

Just like you say, if we want to delete a ConfigMap externally. We just need to call `configMapStore#remove(name)`. And the `configMapStore` in `TestingFlinkKubeClient#Builder` could be passed externally.

If you insist, I could add several `Function`s instead.

Reply:
Either `configMapStore#remove(name)` or passing in `configMapStore` requires the map to be accessed from both internal and external of the testing client. Then we would also need to deal with the concurrent access issue.

I would vote for the `Function` approach, to avoid adding maintaining overhead for future efforts.

Reply:
Hmm. I know your concerns now. Then I will keep the `configMapStore` in `TestingFlinkKubeClient` internally. And add `Functions` for the flexibility, for example, throw an exception when `createConfigMap`.